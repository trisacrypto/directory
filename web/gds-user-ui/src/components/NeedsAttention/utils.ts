export const enum AttentionAction {
  'VERIFY_EMAILS' = 'VERIFY_EMAILS',
  'VERIFY_PHONES' = 'VERIFY_PHONES',
  'VERIFY_ADDRESSES' = 'VERIFY_ADDRESSES',
  'VERIFY_IDENTITIES' = 'VERIFY_IDENTITIES',
  'VERIFY_DOCUMENTS' = 'VERIFY_DOCUMENTS',
  'VERIFY_ALL' = 'VERIFY_ALL',
  'VERIFY_NONE' = 'VERIFY_NONE'
}
