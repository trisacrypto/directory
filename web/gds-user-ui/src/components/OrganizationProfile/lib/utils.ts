export const StatusCheck = (status: string) => {
  switch (status) {
    case 'NO_VERIFICATION':
      return 'Not Verified';
    case 'PENDING_REVIEW':
      return 'Pending Review';
    case 'VERIFIED':
      return 'Verified';
    case 'REJECTED':
      return 'Rejected';
    default:
      return 'Pending Registration';
  }
};
