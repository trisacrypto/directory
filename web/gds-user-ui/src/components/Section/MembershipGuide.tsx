import { Flex, Heading, Stack, VStack, Text, Box } from '@chakra-ui/react';
import { t } from '@lingui/macro';
import { Trans } from '@lingui/react';
import Footer from 'components/Footer/LandingFooter';
import LandingHeader from 'components/Header/LandingHeader';
import MembershipGuideCard from 'components/MembershipGuideCard';
import { NavBar } from 'components/Navbar/Landing/Nav';
import React from 'react';

const MembershipGuideText = [
  {
    stepNumber: 1,
    header: t`create your account`,
    description: t`Create your TRISA account with your VASP email address. Add collaborators in your organization.`,
    buttonText: t`Create Account`
  },
  {
    stepNumber: 2,
    header: t`complete VASP verification`,
    description: t`Complete the multi-part TRISA verification form and due diligence process. Once approved, gain access to the Testnet.`,
    buttonText: t`Learn More`
  },
  {
    stepNumber: 3,
    header: t`Integrate and Comply`,
    description: t`Set up your TRISA node or integrate with a 3rd-party Travel Rule solution. Complete testing and move to production.`,
    buttonText: t`Learn More`
  }
];

const MembershipGuide = () => {
  return (
    <Box minHeight="100vh">
      <LandingHeader />
      <section>
        <Stack>
          <Flex
            bgGradient="linear-gradient(90.17deg, rgba(35, 167, 224, 0.85) 3.85%, rgba(27, 206, 159, 0.55) 96.72%);"
            color="white"
            width="100%"
            minHeight={286}
            justifyContent="center"
            direction="column"
            paddingY={{ base: 12, md: 16 }}
            px="1rem"
            fontSize={'xl'}>
            <Stack textAlign={'center'} color="white" spacing={{ base: 3 }}>
              <VStack spacing={1}>
                <Heading
                  fontWeight={700}
                  fontFamily="Open Sans, sans-serif !important"
                  fontSize={{ md: '4xl', sm: '2xl' }}
                  color="#fff">
                  <Trans id="Welcome to TRISA’s network of Certified VASPs.">
                    Welcome to TRISA’s network of Certified VASPs.
                  </Trans>
                </Heading>
                <Text as="p" mt={2}>
                  <Trans id="Learn about the three-step certificaton process">
                    Learn about the three-step certificaton process
                  </Trans>
                </Text>
                <Text as="p" mt={2}>
                  <Trans id="and create your account today.">and create your account today.</Trans>
                </Text>
              </VStack>
            </Stack>
          </Flex>
          <Stack
            justifyContent={'center'}
            alignItems={['center', 'stretch']}
            spacing={10}
            direction={['column', 'row']}
            py={'2rem'}>
            {MembershipGuideText.map(({ stepNumber, header, description, buttonText }) => (
              <React.Fragment key={stepNumber}>
                <MembershipGuideCard
                  stepNumber={stepNumber}
                  header={header}
                  description={description}
                  buttonText={buttonText}
                />
              </React.Fragment>
            ))}
          </Stack>
        </Stack>
      </section>
      <Footer />
    </Box>
  );
};

export default MembershipGuide;
