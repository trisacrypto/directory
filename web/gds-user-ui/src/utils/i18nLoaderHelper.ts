import { i18n } from '@lingui/core';
import { en, fr, de, zh, ja, ru } from 'make-plural/plurals';
import { Locales } from 'types/type';

export const DEFAULT_LOCALE: Locales = 'en';

export async function dynamicActivate(locale: string) {
  const { messages } = await import(`../locales/${locale}/messages`);

  i18n.load(locale, messages);
  i18n.loadLocaleData({
    en: { plurals: en },
    de: { plurals: de },
    fr: { plurals: fr },
    zh: { plurals: zh },
    ja: { plurals: ja }
  });
  i18n.activate(locale);
}
