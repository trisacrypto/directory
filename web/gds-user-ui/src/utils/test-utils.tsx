import { ChakraProvider } from '@chakra-ui/react';
import { render as rtlRender, RenderOptions as RTLRenderOptions } from '@testing-library/react';
import React from 'react';
import { FormProvider, useForm } from 'react-hook-form';
import { Provider } from 'react-redux';
import customTheme from './theme';
import rootReducer from 'application/store/rootReducer';
import { configureStore, Store } from '@reduxjs/toolkit';

type RenderOptions = {
  rhfDefaultValues?: { [x: string]: any };
  store?: Store;
  preloadedState?: StoreState;
  renderOptions?: Omit<RTLRenderOptions, 'wrapper'>;
};

type StoreState = {
  [x: string]: any;
};

export function render(
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  {
    rhfDefaultValues,
    preloadedState,
    store = configureStore({ reducer: rootReducer, preloadedState }),
    ...renderOptions
  }: RenderOptions = {}
) {
  function Wrapper({ children }: { children: React.ReactNode }) {
    const methods = useForm({ defaultValues: rhfDefaultValues });
    return (
      <Provider store={store}>
        <ChakraProvider theme={customTheme}>
          <FormProvider {...methods}>{children}</FormProvider>
        </ChakraProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
