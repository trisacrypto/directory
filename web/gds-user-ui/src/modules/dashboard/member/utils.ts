import { DirectoryType, VaspDirectoryEnum, DirectoryTypeEnum } from './memberType';
import { convertToCVS, downloadCSV } from 'utils/utils';
import { t } from '@lingui/macro';

export const memberTableHeader = [
  {
    key: 'name',
    label: t`Member Name`
  },
  {
    key: 'first_listed',
    label: t`Joined`
  },
  {
    key: 'last_updated',
    label: t`Last Updated`
  },
  {
    key: 'registered_directory',
    label: t`Network`
  },
  {
    key: 'status',
    label: t`Status`
  }
];
export const getVaspDirectory = (dir: DirectoryType) => {
  return dir === DirectoryTypeEnum.TESTNET ? VaspDirectoryEnum.TESTNET : VaspDirectoryEnum.MAINNET;
};

export const getVaspNetwork = (dir: any) => {
  switch (dir) {
    case (VaspDirectoryEnum.TESTNET, VaspDirectoryEnum.TESTNET_DEV):
      return 'TestNet';
    case (VaspDirectoryEnum.MAINNET, VaspDirectoryEnum.MAINNET_DEV):
      return 'MainNet';
    default:
      return 'MainNet';
  }
};

export const getVaspStatus = (status: number) => {
  switch (status) {
    case 1:
      return t`NO VERIFICATION`;
    case 2:
      return t`SUBMITTED`;
    case 3:
      return t`EMAIL VERIFIED`;
    case 4:
      return t`PENDING REVIEW`;
    case 5:
      return t`REVIEWED`;
    case 6:
      return t`VERIFIED`;
    case 7:
      return t`REJECTED`;
    case 8:
      return t`APPEALED`;
    case 9:
      return t`ERRORED`;
    default:
      return t`NO VERIFICATION`;
  }
};

export const downloadMembers2CVS = (member: any) => {
  const m = member.map((item: any) => {
    return {
      ...item,
      status: getVaspStatus(item.status),
      registered_directory: getVaspNetwork(item.registered_directory)
    };
  });
  const memberCsv = convertToCVS(m, memberTableHeader as ITableHeader[]);
  downloadCSV(memberCsv, 'members');
};

