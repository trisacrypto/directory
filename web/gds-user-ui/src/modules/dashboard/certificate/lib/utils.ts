import { basicDetailsValidationSchema } from 'modules/dashboard/certificate/lib/basicDetailsValidationSchema';
import { legalPersonValidationSchemam } from 'modules/dashboard/certificate/lib/legalPersonValidationSchema';
import { contactsValidationSchema } from 'modules/dashboard/certificate/lib/contactsValidationSchema';
import { trisaImplementationValidationSchema } from 'modules/dashboard/certificate/lib/trisaImplementationValidationSchema';
import { trixoQuestionnaireValidationSchema } from 'modules/dashboard/certificate/lib/trixoQuestionnaireValidationSchema';

export const isBasicDetailsCompleted = async (data: any) => {
  try {
    const r = await basicDetailsValidationSchema.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    return false;
  }
};

export const isLegalPersonCompleted = async (data: any) => {
  try {
    const r = await legalPersonValidationSchemam.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    console.log('legal person validation failed', err);
    return false;
  }
};

export const isContactsCompleted = async (data: any) => {
  try {
    const r = await contactsValidationSchema.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    console.log('contacts validation failed', err);
    return false;
  }
};

export const isTrisaImplementationCompleted = async (data: any) => {
  try {
    await trisaImplementationValidationSchema.validate(data);
    return true;
  } catch (err) {
    return false;
  }
};

export const isTrixoQuestionnaireCompleted = async (data: any) => {
  try {
    await trixoQuestionnaireValidationSchema.validate(data);
    return true;
  } catch (err) {
    return false;
  }
};
