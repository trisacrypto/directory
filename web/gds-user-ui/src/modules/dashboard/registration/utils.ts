import { getDefaultValue } from 'components/BasicDetailsForm/validation';
import { getRegistrationDefaultValues } from 'modules/dashboard/certificate/lib';
import {
  postRegistrationData,
  getRegistrationData,
  getSubmissionStatus
} from 'modules/dashboard/registration/service';
import { handleError } from 'utils/utils';

export const postRegistrationValue = (data: any) => {
  console.log('[postRegistrationValue]', data);
  return new Promise((resolve, reject) => {
    postRegistrationData(data)
      .then((res) => {
        console.log('[postRegistrationData]', res);
        if (res.status === 204) {
          resolve(res);
        } else {
          reject(res);
        }
      })
      .catch((err) => {
        console.log('[postRegistrationData]', err);
        reject(err);
      });
  });
};
export const getRegistrationDefaultValue = async () => {
  try {
    const regData = await getRegistrationData();
    if (regData.status === 200 && Object.keys(regData.data).length > 0) {
      return regData.data;
    }
    const defaultValue: any = localStorage.getItem('certificateForm');
    if (defaultValue) {
      const val = JSON.parse(defaultValue);
      const postData = await postRegistrationData(val);
      if (postData.status === 204) {
        const getData = await getRegistrationData();
        localStorage.removeItem('certificateForm');
        return getData.data;
      }
    }
    const v = getRegistrationDefaultValues();
    await postRegistrationValue(v);
    return getRegistrationDefaultValues();
  } catch (err: any) {
    handleError(err, 'failed to get registration data');
  }
};

export const setRegistrationDefaultValue = () => {
  console.log('[setRegistrationDefaultValue]');
  const defaultValue: any = getRegistrationDefaultValues();
  return new Promise((resolve, reject) => {
    postRegistrationData(defaultValue)
      .then((res) => {
        // console.log('[default postRegistration value]', res);
        if (res.status === 204) {
          resolve(res);
        } else {
          reject(res);
        }
      })
      .catch((err) => {
        handleError(err, 'failed to post registration value');
        reject(err);
      });
  });
};

// get registration data from backend and download the file
export const downloadRegistrationData = async () => {
  try {
    const regData = await getRegistrationData();
    if (regData.status === 200 && Object.keys(regData.data).length > 0) {
      const blob = new Blob([JSON.stringify(regData.data)], { type: 'application/json' });
      const url = window.URL.createObjectURL(blob);
      const link = document.createElement('a');
      link.href = url;
      link.setAttribute('download', 'registration.json');
      document.body.appendChild(link);
      link.click();
    }
  } catch (err: any) {
    handleError(err, 'failed to get registration data');
  }
};

// load default stepper

export const getDefaultStepper = async () => {
  try {
    const [regData, regStatus] = await Promise.all([getRegistrationData(), getSubmissionStatus()]);
    console.log('[regStatus]', regStatus.data);
    if (regData.status === 200 && Object.keys(regData.data).length > 0) {
      return {
        currentStep: regData.data.state.current,
        steps: regData.data.state.steps,
        lastStep: null,
        hasReachSubmitStep: regData.data.state.ready_to_submit,
        testnetSubmitted: regStatus?.data?.testnetSubmitted || false,
        mainnetSubmitted: regStatus?.data?.mainnetSubmitted || false
      };
    }
    const defaultValue: any = {
      current: 1,
      steps: [
        {
          key: 1,
          status: 'progress'
        }
      ]
    };
    // update registrations state object
    const postData = await postRegistrationData({ state: defaultValue });
    if (postData.status === 204) {
      const getData = await getRegistrationData();
      return {
        currentStep: getData.data.state.current,
        steps: getData.data.state.steps,
        lastStep: null,
        hasReachSubmitStep: false,
        testnetSubmitted: false,
        mainnetSubmitted: false
      };
    }
  } catch (err: any) {
    handleError(err, 'failed to get stepper data');
  }
};

// // load default stepper without async call
// export const loadDefaultStepperSync = () => {
//   return Promise.resolve({
//     currentStep: 1,
//     steps: [
//       {
//         key: 1,
//         status: 'progress'
//       }
//     ],
//     lastStep: null,
//     hasReachSubmitStep: false
//   });
// };

// set stepper data
export const getRegistrationAndStepperData = async () => {
  try {
    const [regData, regStatus] = await Promise.all([
      getRegistrationDefaultValue(),
      getSubmissionStatus()
    ]);
    if (regData) {
      const response: any = {
        registrationData: regData,
        stepperData: {
          currentStep: regData?.state?.current || 1,
          steps: regData?.state?.steps || [{ key: 1, status: 'progress' }],
          lastStep: null,
          hasReachSubmitStep: regData?.state?.ready_to_submit || false,
          testnetSubmitted: !!regStatus?.data?.testnet_submitted,
          mainnetSubmitted: !!regStatus?.data?.mainnet_submitted
        }
      };
      return response;
    }
  } catch (err: any) {
    handleError(err, 'failed to get stepper data');
  }
};
