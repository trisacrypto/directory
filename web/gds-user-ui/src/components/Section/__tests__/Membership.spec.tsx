import { dynamicActivate } from 'utils/i18nLoaderHelper';
import { act, render, screen } from 'utils/test-utils';
import MembershipGuide from '../MembershipGuide';

describe('<MembershipGuide />', () => {
  beforeAll(() => {
    act(() => {
      dynamicActivate('en');
    });
  });

  it('should match snapshot', () => {
    const { container } = render(<MembershipGuide />);

    expect(container).toMatchSnapshot();
  });
});
