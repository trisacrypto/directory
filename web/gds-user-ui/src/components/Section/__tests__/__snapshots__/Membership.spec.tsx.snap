// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MembershipGuide /> should match snapshot 1`] = `
<div>
  <div
    class="css-vooagt"
  >
    <div
      class="css-1pwxvtk"
    >
      <div
        class="chakra-container css-1a3ltpp"
      >
        <div
          class="css-104xflu"
        >
          <div
            class="css-1uuxm7r"
          >
            <div
              class="css-0"
            >
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                <a
                  class="chakra-link css-htwawm"
                >
                  <div
                    class="css-1p293e5"
                  >
                    <img
                      alt="Trisa logo"
                      src="trisa_logo.svg"
                    />
                  </div>
                </a>
              </a>
            </div>
            <div
              class="css-1ydjxjn"
            >
              <svg
                viewBox="0 0 20 20"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Menu
                </title>
                <path
                  d="M0 3h20v2H0V3zm0 6h20v2H0V9zm0 6h20v2H0v-2z"
                  fill="black"
                />
              </svg>
            </div>
            <div
              class="chakra-stack css-sz9ruu"
            >
              <div
                class="chakra-stack css-1p6wuvz"
              >
                <div
                  class="chakra-select__wrapper css-aq64ov"
                >
                  <select
                    class="chakra-select css-k7r2wc"
                  >
                    <option
                      value="en"
                    >
                      ðŸ‡ºðŸ‡¸ EN
                    </option>
                    <option
                      value="fr"
                    >
                      ðŸ‡«ðŸ‡· FR
                    </option>
                    <option
                      value="de"
                    >
                      ðŸ‡©ðŸ‡ª DE
                    </option>
                    <option
                      value="zh"
                    >
                      ðŸ‡¨ðŸ‡³ ZH
                    </option>
                    <option
                      value="ja"
                    >
                      ðŸ‡¯ðŸ‡µ JA
                    </option>
                    <option
                      value="en-dh"
                    >
                      - DH
                    </option>
                  </select>
                  <div
                    class="chakra-select__icon-wrapper css-zj2dqk"
                  >
                    <svg
                      aria-hidden="true"
                      class="chakra-select__icon"
                      focusable="false"
                      role="presentation"
                      style="width: 1em; height: 1em; color: currentColor;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="chakra-text css-dme2i4"
              >
                <a
                  href="/#about"
                >
                  <a
                    class="chakra-link css-htwawm"
                  >
                    About TRISA
                  </a>
                </a>
              </div>
              <div
                class="chakra-text css-dme2i4"
                data-testid="documentation"
              >
                <a
                  class="chakra-link css-htwawm"
                  href="https://trisa.dev/en"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Documentation
                </a>
              </div>
              <div
                class="chakra-stack css-n21gh5"
              >
                <a
                  class=""
                  href="/auth/login"
                >
                  <button
                    class="chakra-button css-1t0rcyz"
                    type="button"
                  >
                    Login
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <section>
      <div
        class="chakra-stack css-n21gh5"
      >
        <div
          class="css-1u814me"
        >
          <div
            class="chakra-stack css-1vqyy53"
          >
            <div
              class="chakra-stack css-126kq63"
            >
              <h2
                class="chakra-heading css-1y1itts"
              >
                Welcome to TRISAâ€™s network of Certified VASPs.
              </h2>
              <p
                class="chakra-text css-rltemf"
              >
                Learn about the three-step process to become a member and verified VASP.
              </p>
              <p
                class="chakra-text css-rltemf"
              >
                Create your account today.
              </p>
            </div>
          </div>
        </div>
        <div
          class="chakra-stack css-1kojfy"
        >
          <div
            class="css-wsis5l"
          >
            <div
              class="chakra-stack css-ly7s1u"
            >
              <div
                class="chakra-stack css-owjkmg"
              >
                <p
                  class="chakra-text css-wz7pgp"
                  data-testid="step"
                >
                  Step 
                  1
                </p>
                <p
                  class="chakra-text css-8cgwqi"
                  data-testid="header"
                >
                  create your account
                </p>
                <p
                  class="chakra-text css-0"
                  data-testid="description"
                >
                  Create your TRISA account with your VASP email address. Add collaborators in your organization.
                </p>
              </div>
              <div
                class="css-14d2w9n"
              >
                <button
                  class="chakra-button css-vx0ruo"
                  type="button"
                >
                  Create Account
                </button>
              </div>
            </div>
          </div>
          <div
            class="css-wsis5l"
          >
            <div
              class="chakra-stack css-ly7s1u"
            >
              <div
                class="chakra-stack css-owjkmg"
              >
                <p
                  class="chakra-text css-wz7pgp"
                  data-testid="step"
                >
                  Step 
                  2
                </p>
                <p
                  class="chakra-text css-8cgwqi"
                  data-testid="header"
                >
                  complete VASP verification
                </p>
                <p
                  class="chakra-text css-0"
                  data-testid="description"
                >
                  Complete the multi-part TRISA verification form and due diligence process. Once approved, gain access to the Testnet and MainNet.
                </p>
              </div>
              <div
                class="css-14d2w9n"
              >
                <button
                  class="chakra-button css-vx0ruo"
                  type="button"
                >
                  Learn More
                </button>
              </div>
            </div>
          </div>
          <div
            class="css-wsis5l"
          >
            <div
              class="chakra-stack css-ly7s1u"
            >
              <div
                class="chakra-stack css-owjkmg"
              >
                <p
                  class="chakra-text css-wz7pgp"
                  data-testid="step"
                >
                  Step 
                  3
                </p>
                <p
                  class="chakra-text css-8cgwqi"
                  data-testid="header"
                >
                  Integrate and Comply
                </p>
                <p
                  class="chakra-text css-0"
                  data-testid="description"
                >
                  Set up your TRISA node or integrate with a 3rd-party Travel Rule solution. Complete testing and move to production.
                </p>
              </div>
              <div
                class="css-14d2w9n"
              >
                <button
                  class="chakra-button css-vx0ruo"
                  type="button"
                >
                  Learn More
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
    <footer
      style="width: 100%;"
    >
      <div
        class="chakra-stack css-1arqzha"
      >
        <div
          class="css-248241"
        >
          <p
            class="chakra-text css-q7d5fr"
          >
            A component of
             
            <a
              class="chakra-link css-1av0s05"
              href="https://trisa.io"
            >
              the TRISA architecture
            </a>
             
            for Cryptocurrency Travel Rule compliance.
          </p>
          <p
            class="chakra-text css-q7d5fr"
          >
            Created and maintained by
             
            <a
              class="chakra-link css-1av0s05"
              href="https://rotational.io"
            >
               
              Rotational Labs
            </a>
             
            in partnership with
             
            <a
              class="chakra-link css-1av0s05"
              href="https://cyphertrace.com"
            >
               
              CipherTrace
            </a>
             
            on behalf of
             
            <a
              class="chakra-link css-1av0s05"
              href="https://trisa.io"
            >
              TRISA
            </a>
             
          </p>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
